lib/ansible/modules/notification/bearychat.py

```diff
@@ -0,0 +1,186 @@
+        color: "#ffffff"
+        images:
+          - http://example.com/index.png
+"""
+
+RETURN = """
+msg:
+    description: execution result
+    returned: success
+    type: string
+    sample: "OK"
+"""
+
+try:
+    from ansible.module_utils.six.moves.urllib.parse import urlparse, urlunparse
+    HAS_URLPARSE = True
+except:
+    HAS_URLPARSE = False
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
+
+def build_payload_for_bearychat(module, text, markdown, channel, attachments):
+    payload = {}
+    if text is not None:
+        payload['text'] = text
+    if markdown is not None:
+        payload['markdown'] = markdown
+    if channel is not None:
+        payload['channel'] = channel
+    if attachments is not None:
+        payload.setdefault('attachments', []).extend(
+            build_payload_for_bearychat_attachment(
+                module, item.get('title'), item.get('text'), item.get('color'),
+                item.get('images'))
+            for item in attachments)
+    payload = 'payload=%s' % module.jsonify(payload)
+    return payload
+
+
+def build_payload_for_bearychat_attachment(module, title, text, color, images):
+    attachment = {}
+    if title is not None:
+        attachment['title'] = title
+    if text is not None:
+        attachment['text'] = text
+    if color is not None:
+        attachment['color'] = color
+    if images is not None:
+        target_images = attachment.setdefault('images', [])
+        if not isinstance(images, (list, tuple)):
+            images = [images]
+        for image in images:
+            if isinstance(image, dict) and 'url' in image:
+                image = {'url': image['url']}
+            elif hasattr(image, 'startswith') and image.startswith('http'):
+                image = {'url': image}
+            else:
+                module.fail_json(
+                    msg="BearyChat doesn't have support for this kind of "
+                        "attachment image")
+            target_images.append(image)
+    return attachment
+
+
+def do_notify_bearychat(module, url, payload):
+    if not HAS_URLPARSE:
+        module.fail_json(msg='urlparse is not installed')
+    response, info = fetch_url(module, url, data=payload)
+    if info['status'] != 200:
+        url_info = urlparse(url)
+        obscured_incoming_webhook = urlunparse(
+            (url_info.scheme, url_info.netloc, '[obscured]', '', '', ''))
+        module.fail_json(
+            msg=" failed to send %s to %s: %s" % (
+                payload, obscured_incoming_webhook, info['msg']))
+
+
+def main():
+    module = AnsibleModule(argument_spec={
+        'url': dict(type='str', required=True, no_log=True),
+        'text': dict(type='str', required=False, default=None),
+        'markdown': dict(default='yes', type='bool'),
+        'channel': dict(type='str', default=None),
+        'attachments': dict(type='list', required=False, default=None),
+    })
```

Comment:
Required=false are implicit could you remove them ?
Default=None too in case of string